core/shared/src/main/scala/zio/Chunk.scala

```diff
@@ -928,12 +1008,79 @@ object Chunk {
 
     override val length: Int = array.length
 
+    override def toArray[A1 >: A](n: Int, dest: Array[A1]): Unit =
+      Array.copy(array, 0, dest, n, length)
+  }
+
+  private object Arr {
+    def apply[A](array: Array[A]): Arr[A] = {
+      val clss = array.getClass().getComponentType()
+      val res =
+        if (clss == java.lang.Boolean.TYPE) BooleanArr(array.asInstanceOf[Array[Boolean]])
```

Comment:
`java.lang.Boolean/Byte/etc.` are the boxed versions of primitives. Are you sure this isn't boxing the data in the array?

Reply:
https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html#TYPE

https://repl.it/repls/StridentHealthyVirus

Reply:
Aha! TIL! Name was funny ;)

Reply:
Just Java things.